### PR TITLE
Fix file extraction

### PR DESCRIPTION
**Pull request by DixitPatel**
This change fixes the file extraction for compressed files. Only gzip file decompression is supported currently.
Steps using decompression :

    Mutation Data Step (MAF files)

### CODE REVIEW

File: src/main/java/org/cbio/gdcpipeline/listener/MutationStepListener.java [hunk 1]
```diff
@@ -81,16 +81,19 @@ public ExitStatus afterStep(StepExecution stepExecution) {
...
 
     public List<File> getMutationFileList() {
         List<File> fileList = CommonDataUtil.getFileList(gdcFileMetadatas, CommonDataUtil.GDC_TYPE.MUTATION, sourceDir);
-        List<File> mutationFileList = new ArrayList<>();
-        for (File file : fileList) {
-            File maf =new File(sourceDir,file.getName().replace(".gz", ""));
-            mutationFileList.add(maf);
+        List<File> mutationFileList = null;
+        try {
+            mutationFileList = CommonDataUtil.extractCompressedFiles(fileList);
```
Comment (zheins) [inline src/main/java/org/cbio/gdcpipeline/listener/MutationStepListener.java]: I kind of feel like `getFileList` should automatically extract the files. You shouldn't have to make two calls into the `CommonDataUtil`.

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -18,5 +18,11 @@ mutation.data.file.prefix=data_
 mutation.metadata.file=meta_mutation.txt
 mutation.default.merged.maf.file=data_extended.maf
 
+#################################### Genome Nexus ###################################################
```
Comment (zheins) [inline src/main/resources/application.properties]: Remove this

File: src/main/java/org/cbio/gdcpipeline/listener/MutationStepListener.java (hunk 1 above)
Comment (DixitPatel) [reply src/main/java/org/cbio/gdcpipeline/listener/MutationStepListener.java]: Then either I make the extractCompressedFile call dependent on Hits parameter or I add extra logic to check inside getFileList to see if a file is compressed and extract it accordingly. What do you think ?

File: src/main/resources/application.properties (hunk 2 above)
Comment (DixitPatel) [reply src/main/resources/application.properties]: Are you talking about genome nexus or mutation config ? We need both right..

Comment (zheins) [reply src/main/resources/application.properties]: It shouldn't be in this PR

Comment (n1zea144) [reply src/main/resources/application.properties]: @DixitPatel: I think what @zheins is saying that this PR is not about variant annotation, but fixes to file extraction / compression.  Can you move these properties to PR #12?

**Comment by n1zea144** [comment]
Can you squash the three commits in this PR into a single one?

File: src/main/java/org/cbio/gdcpipeline/util/CommonDataUtil.java [hunk 3]
```diff
@@ -87,6 +115,96 @@ public String toString(){
...
+            }
+        }
+        return extracted;
+    }
+
+    private static boolean isCompressedFile(File extractFile) {
+        return extractFile.getName().endsWith(COMPRESSION_FORMAT.GZIP.toString());
+    }
+
+    public static void deleteTempDir() {
```
Comment (n1zea144) [inline src/main/java/org/cbio/gdcpipeline/util/CommonDataUtil.java]: I would use an open source library to perform IO operations like this.  I would suggest [Apache Commons FileUtils](https://commons.apache.org/proper/commons-io/javadocs/api-2.5/index.html?org/apache/commons/io/FileUtils.html)

File: src/main/java/org/cbio/gdcpipeline/util/CommonDataUtil.java [hunk 4]
```diff
@@ -87,6 +115,96 @@ public String toString(){
...
+            } catch (Exception e) {
+                if (LOG.isWarnEnabled()) {
+                    LOG.warn(" Temp directory could not be deleted : " + temp_dir.getAbsolutePath());
+                }
+                e.printStackTrace();
+            }
+        }
+    }
+
+    private static File createTempDirectory() throws Exception {
```
Comment (n1zea144) [inline src/main/java/org/cbio/gdcpipeline/util/CommonDataUtil.java]: I would use an open source library to perform IO operations like this.  I would suggest [Apache Commons FileUtils](https://commons.apache.org/proper/commons-io/javadocs/api-2.5/index.html?org/apache/commons/io/FileUtils.html)

File: src/main/java/org/cbio/gdcpipeline/util/CommonDataUtil.java [hunk 5]
```diff
@@ -87,6 +115,96 @@ public String toString(){
...
+            if (LOG.isErrorEnabled()) {
+                LOG.error("Temp directory already exists. Deleting directory and its contents :" + tmp_dir.getAbsolutePath());
+            }
+            deleteDir(tmp_dir);
+        }
+        tmp_dir.mkdir();
+        return tmp_dir;
+    }
+
+    private static void deleteDir(File dir) throws Exception {
```
Comment (n1zea144) [inline src/main/java/org/cbio/gdcpipeline/util/CommonDataUtil.java]: I would use an open source library to perform IO operations like this.  I would suggest [Apache Commons FileUtils](https://commons.apache.org/proper/commons-io/javadocs/api-2.5/index.html?org/apache/commons/io/FileUtils.html)

**Comment by ao508** [comment]
@DixitPatel We are continuing development of this pipeline. Would you mind resolving these conflicts when you have a chance?

@n1zea144 had also suggested using an open source library for the directory deletion. I created an issue for it here: #16  if you can't get to it

**Pull request closed without merging**